src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -14,71 +14,85 @@
 
     private ReadFilterLibrary(){ /*no instance*/ }
 
-    /**
-     * local classes for static read filters
-     */
+    /** Do not filter out any read. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class AllowAllReadsReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){return true;}}
 
+    /** Filter out reads containing skipped region from the reference (CIGAR strings with 'N' operator). */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     //Note: do not call getCigar to avoid creation of new Cigar objects
     public static class CigarContainsNoNOperator extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){
             return ! CigarUtils.containsNOperator(read.getCigarElements());}}
 
+    /** Keep only reads that are first of pair. */
     @DocumentedFeature(groupName=HelpConstants.DOC_CAT_READFILTERS, groupSummary=HelpConstants.DOC_CAT_READFILTERS_SUMMARY)
     public static class FirstOfPairReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test (final GATKRead read) {
             return read.isFirstOfPair();}}
 
+    /**
+     * Keep only reads containing good CIGAR strings:
+     * <ul>
+     *     <li>Valid regarding the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications.</a></li>
+     *     <li>Does not start or end with deletions (with or without preceding clips) </li>
+     *     <li>Has not consecutive deletions/insertions operators.</li>
```

Comment:
@vdauwera I haven't been keeping up on these changes, so I'm adding a few random thoughts as I look at these changes. In this example, if we include an html list in the javadoc like this, and don't include a summary in the DocumentedFeature annotation, the doc system will put this javadoc, including the list, in the index page in place of the summary:


<img width="864" alt="screen shot 2017-06-24 at 11 41 15 am" src="https://user-images.githubusercontent.com/10062863/27509779-34d97658-58d2-11e7-831d-e7676088268d.png">
Its only doing this because there is no summary line (and in this case, because its a read filter, there are no CommandLineProgramProperties, which have a summary). There are some notes about this [here](https://github.com/broadinstitute/gatk/wiki/GATK4-Documentation-Guidelines).